lib/cadet/shared_programs/shared_program.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Cadet.SharedPrograms.SharedProgram do
+  @moduledoc """
+  Contains methods for storing frontend programs to database with uuid.
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "shared_programs" do
+    field(:data, :map)
+    field(:uuid, Ecto.UUID)
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(shared_program, attrs) do
+    shared_program
```

Comment:
Missing definitions for required and optional fields (take a look at the other models in the codebase to see how they're implemented).